test/int/adapter.test.ts

```diff
@@ -72,18 +85,49 @@ suite('Chrome Debug Adapter etc', function () {
         });
 
         test('should stop on debugger statement in http://localhost', () => {
-            const testProjectRoot = path.join(DATA_ROOT, 'intervalDebugger');
             const breakFile = path.join(testProjectRoot, 'src/app.ts');
             const DEBUGGER_LINE = 2;
 
-            server = createServer({ root: testProjectRoot });
-            server.listen(7890);
-
             return Promise.all([
                 dc.configurationSequence(),
                 dc.launch({ url: 'http://localhost:7890', webRoot: testProjectRoot }),
                 dc.assertStoppedLocation('debugger_statement', { path: breakFile, line: DEBUGGER_LINE } )
             ]);
         });
+
+        test('Should attach to existing instance of chrome and break on debugger statement', async () => {
+            const breakFile = path.join(testProjectRoot, 'src/app.ts');
+            const DEBUGGER_LINE = 2;
+
+            const browser = await puppeteer.launch({ headless: false, args: ['http://localhost:7890', '--remote-debugging-port=7777'] });
+
+            await Promise.all([
+                dc.configurationSequence(),
+                dc.initializeRequest().then(_ => {
+                    return dc.attachRequest(<IAttachRequestArgs>{ url: 'http://localhost:7890', port: 7777, webRoot: testProjectRoot });
+                }),
+                dc.assertStoppedLocation('debugger_statement', { path: breakFile, line: DEBUGGER_LINE } )
+            ]);
+
+            await browser.close();
+        });
+
+        test('Should throw error when launching if chrome debug port is in use', async () => {
+            // browser already launched to the default port, and navigated away from about:blank
+            const browser = await puppeteer.launch({ headless: false, args: ['http://localhost:7890', '--remote-debugging-port=9222'] });
+            try {
+                await Promise.all([
+                    dc.configurationSequence(),
+                    dc.launch({ url: 'http://localhost:7890', timeout: 2000, webRoot: testProjectRoot }),
+                ]);
+            } catch (err) {
+                expect(err.message).to.satisfy( (x: string) => x.startsWith('Cannot connect to runtime process, timeout after 2000 ms'));
```

Comment:
Will this test fail if dc.launch somehow succeeds?

Reply:
Yes. Right now since we're just baselining V1 behavior, we are expecting this to throw an exception. Once we decide it's an appropriate time to fix this, then we can change the test accordingly. I'll add a note in the test that explains this so that it's clear what we're expecting from the test